python/pyspark/ml/param/__init__.py

```diff
@@ -247,7 +248,27 @@ def _set(self, **kwargs):
         Sets user-supplied params.
         """
         for param, value in kwargs.items():
-            self._paramMap[getattr(self, param)] = value
+            p = getattr(self, param)
+            if (p.expectedType is not None):
+                if p.expectedType is None or type(value) == p.expectedType or value is None:
```

Comment:
I believe the `p.expectedType is None` check in the second if statement is frivolous. Couldn't this be condensed into a single if else like:

```python
if p.expectedType is not None and type(value) != p.expectedType and value is not None:
    try:
        ...
else:
    self._paramMap[getattr(self, param)] = value
```